api/v1alpha1/gatewayclass_types.go

```diff
@@ -26,22 +26,95 @@ import (
 // +k8s:openapi-gen=true
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// GatewayClass describes a class of gateways av
-// for defining access to their routed services. GatewayClass allow a
+// GatewayClass describes a class of gateways available to the user
+// for creating Gateway resources.
 //
-// GatewayClass is a non-namespaced resource.
+// GatewayClass is a Cluster level resource.
+//
+// Support: Core.
 type GatewayClass struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// Controller is a domain/path that denotes which controller is responsible
-	// for this class. Example: "acme.io/gateway-controller".
-	Controller string `json:"controller" protobuf:"bytes,2,opt,name=controller"`
+	// Spec for this GatewayClass.
+	Spec GatewayClassSpec `json:"spec,omitempty"`
+	// Status of the GatewayClass.
+	Status GatewayClassStatus `json:"status,omitempty"`
+}
+
+// GatewayClassSpec reflects the configuration of a class of Gateways.
+type GatewayClassSpec struct {
+	// Controller is a domain/path string that indicates the
+	// controller is responsible for this class. Example:
```

Comment:
Nit: 
```suggestion
	// controller that is responsible for this class. Example:
```

Reply:
Should we call out additional details for this field?

1. Is this field mutable?
2. Should we impose a naming standard such as:
    - [a-z].[a-z]/[a-z]
    - no special characters other than "." or "-"
    - or reference an rfc as a naming standard?
3. Can any vendor pick a name or should the name be registered?
4. Why is gateway class required when ingress class is optional for `Ingress`?

Reply:
1. No, adding that as part of the spec.
2. Adding reference to Kubernetes name: 
```
	// This field is not mutable and cannot be empty. The format
	// of this field is DOMAIN "/" PATH, where DOMAIN and PATH are
	// valid Kubernetes names
	// (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).
```
3. There is no "official registration" for these kinds of names as I know. My thought is that we should follow the experience of https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-provisioning.md in so far as what worked vs not.
4. We couldn't figure out a way w/ the Ingress API to make this required due to backwards compatibility issues.